src/jit/lower.cpp

```diff
@@ -4999,7 +5002,12 @@ GenTree* Lowering::LowerSignedDivOrMod(GenTreePtr node)
     if (!varTypeIsFloating(node->TypeGet()))
 #endif // _TARGET_XARCH_
     {
-        next = LowerConstIntDivOrMod(node);
+        // LowerConstIntDivOrMod will return the existing DIV or MOD node if it is not transformed.
+        GenTree* newNode = LowerConstIntDivOrMod(node);
+        if (newNode != node)
```

Comment:
If seems that `if ((node->OperGet() == GT_DIV) || (node->OperGet() == GT_MOD))` below is no longer needed because of this, it can be changed to an assert.